client/src/components/reader/Section/ResourceViewer/Wrapper.js

```diff
@@ -0,0 +1,94 @@
+import React, { PureComponent, PropTypes } from 'react';
+import Viewer from './Viewer';
+import throttle from 'lodash/throttle';
+
+export default class ResourceViewerWrapper extends PureComponent {
+
+  static displayName = "ResourceViewer.Wrapper";
+
+  static propTypes = {
+    resources: PropTypes.array,
+    containerSize: PropTypes.number,
+    currentResources: PropTypes.array,
+    body: PropTypes.object
+  };
+
+  constructor() {
+    super();
+    this.state = {
+      availableResourceMarkers: [],
+      viewableResourceMarkers: []
+    };
+
+    this.updateAvailableResourceMarkers = this.updateAvailableResourceMarkers.bind(this);
+    this.updateViewableResourceMarkers = this.updateViewableResourceMarkers.bind(this);
```

Comment:
These probably don't need to be bound, since they're never called from the context of an event handler. It's OK if they are, but it's not really necessary.